docs/tutorial/spark-tutorial/t-setup.md

```diff
@@ -0,0 +1,141 @@
+
+The resources that will be used during this tutorial are available at [this GitHub repository](https://github.com/welpaolo/ubuntu-summit-2023). Let's clone that repository onto our machine.
+
+```bash
+git clone https://github.com/welpaolo/ubuntu-summit-2023.git
+cd ubuntu-summit-2023
+```
+
+Charmed Spark is developed to be run on top of a Kubernetes cluster. In this tutorial we use [MicroK8s](https://microk8s.io/), which is the simplest production-grade conformant K8s.
+
+Before starting, it's always a good idea to update the packages in the system.
+
+```bash
+sudo apt update
+sudp apt upgrade
+```
+
+MicroK8s can be installed via Snap as
+
+```bash
+sudo snap install microk8s -channel=1.28-strict/stable
+```
+
+Please note that we're installing the strictly confined version of the MicroK8s snap, because the non-confined one no longer works with Juju (starting from Juju 3.x onwards).
+
+Let's configure MicroK8s for use in this tutorial:
+
+```bash
+# Set an alias 'kubectl' that can be used instead of microk8s.kubectl
+sudo snap alias microk8s.kubectl kubectl
+
+# Add the current user into 'microk8s' group
+sudo usermod -a -G microk8s ${USER}
+
+# Create and provide ownership of '~/.kube' directory to current user
+mkdir -p ~/.kube
+sudo chown -f -R ${USER} ~/.kube
+
+# Put the group membership changes into effect
+newgrp microk8s
+```
+
+Once done, the status of the MicroK8s can be verified with
+
+```bash
+microk8s status --wait-ready
+```
+
+Let's generate the Kubernetes configuration file using MicroK8s and write it to `~/.kube/config`
+
+```bash
+microk8s config | tee ~/.kube/config
+```
+
+In order to use Spark use S3 for object storage, we are going to use an S3 compliant object storage library `minio`, an add-on for which is shipped by default in `microk8s` installation. Let's install the addon and a few other addons.
+
+```bash
+sudo microk8s enable hostpath-storage dns rbac storage minio
+
+sudo apt install jq
+IPADDR=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')
+sudo microk8s enable metallb:$IPADDR-$IPADDR
+```
+
+Once you enable `minio` add-on, you will have a MinIO Kubernetes operator running in the K8s cluster. All the resources created for MinIO are created in the `minio-operator` namespace. You can verify these with the following commands:
+
+```bash
+kubectl get namespace minio-operator
+kubectl get deployments -n minio-operator
+kubectl get services -n minio-operator
+kubectl get pods -n minio-operator
+```
+
+We'll use `juju` to deploy `kafka`, `zookeeper`, `prometheus`, Spark history server and a bunch of other applications later to use together with Spark. For that reason, let's install and configure `juju`:
+
+```bash
+sudo snap install juju --channel 3.1/stable
+
+mkdir -p ~/.local/share
+```
```

Comment:
**todo** Rather, I'm not sure we have ever bootstrapped the Juju controller here.

Reply:
Bootstrapping the Juju controller has been added in the Setup page.